core/cache/application.go

```diff
@@ -28,18 +33,43 @@ type Application struct {
 	configHash string
 }
 
-func (m *Application) setDetails(details ApplicationChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.details = details
+func (a *Application) WatchFields(comparitors ...func(*ApplicationDelta) bool) *ApplicationFieldWatcher {
+	w := newApplicationFieldWatcher(comparitors)
+
+	unsub := a.hub.Subscribe(a.topic(applicationFieldChange), w.detailsChange)
+
+	w.tomb.Go(func() error {
```

Comment:
both the subscription and loop function seem like they could be part of the newApplicatinoFieldWatcher.

Reply:
Done.